lighthouse-core/gather/connections/connection.js

```diff
@@ -52,17 +52,21 @@ class Connection {
    * Call protocol methods
    * @template {keyof LH.CrdpCommands} C
    * @param {C} method
+   * @param {string=} sessionId
    * @param {LH.CrdpCommands[C]['paramsType']} paramArgs,
    * @return {Promise<LH.CrdpCommands[C]['returnType']>}
    */
-  sendCommand(method, ...paramArgs) {
+  sendCommand(method, sessionId, ...paramArgs) {
     // Reify params since we need it as a property so can't just spread again.
     const params = paramArgs.length ? paramArgs[0] : undefined;
 
     log.formatProtocol('method => browser', {method, params}, 'verbose');
     const id = ++this._lastCommandId;
-    const message = JSON.stringify({id, method, params});
-    this.sendRawMessage(message);
+    const message = {id, method, params};
+    // @ts-ignore
+    if (sessionId) message.sessionId = sessionId;
```

Comment:
isn't this equivalent to `JSON.stringify({sessionId, id, method, params})` since undefined will drop out?

Reply:
oh, didn't realize that. cool, will change